tokio/src/runtime/io/scheduled_io.rs

```diff
@@ -429,9 +429,10 @@ impl Future for Readiness<'_> {
     fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
         use std::sync::atomic::Ordering::SeqCst;
 
-        let (scheduled_io, state, waiter) = unsafe {
-            let me = self.get_unchecked_mut();
-            (&me.scheduled_io, &mut me.state, &me.waiter)
+        let (scheduled_io, state, waiter) = {
+            // Safety: `Self` is `!Unpin`
+            let me = unsafe { self.get_unchecked_mut() };
+            (me.scheduled_io, &mut me.state, &me.waiter)
```

Comment:
The reason I didn't use `pin_project` to eliminate the `unsafe` block is that `Future`'s implementation already has a lot of unsafe blocks, and eliminating it doesn't significantly reduce the mental burden.